mars/oscar/backends/pool.py

```diff
@@ -864,15 +864,11 @@ async def handle_control_command(self,
                     self.sub_processes[message.address],
                     timeout=timeout,
                     force=force)
-                if self._auto_recover:
-                    self._recover_events[message.address] = asyncio.Event()
                 processor.result = ResultMessage(message.message_id, True,
                                                  protocol=message.protocol)
             elif message.control_message_type == ControlMessageType.wait_pool_recovered:
-                # check the aliveness of sub pool first, in case monitor task haven't found it.
-                if not await self.is_sub_pool_alive(self.sub_processes[message.address]):
-                    if self._auto_recover and message.address not in self._recover_events:
-                        self._recover_events[message.address] = asyncio.Event()
+                if self._auto_recover and message.address not in self._recover_events:
+                    self._recover_events[message.address] = asyncio.Event()
```

Comment:
I think one single event object is enough. Recover event is created during pool initialization and replaced with a new event every time when a monitor loop finishes.